packages/api/shared/mailer.ts

```diff
@@ -1,16 +1,64 @@
-import { createTransport, SendMailOptions } from 'nodemailer';
+import { createTransport } from 'nodemailer';
+import { EmailStatus } from '../prisma/client';
+import { auth } from './auth';
 
 const transporter = createTransport({
   url: process.env['EMAIL_SERVER'],
 });
 
+export interface EmailOptions {
+  userId: string;
+  subject: string;
+  text: string;
+}
+
+export class EmailNotVerifiedError extends Error {
+  constructor(email: string) {
+    super(`We can't send emails to ${email} because it is not verified.`);
+  }
+}
+
+export class MissingEmailAddressError extends Error {
+  constructor(userId: string) {
+    super(
+      `We can't send emails to user with id ${userId} because they are missing an email address.`
+    );
+  }
+}
+
 export default {
   transporter,
-  async sendEmail(email: SendMailOptions) {
-    // TODO: handle email verification
-    const response = await this.transporter.sendMail({
+  /**
+   * Send a transactional email to a user.
+   * @param email - The email to send along with the intended user.
+   * @param force - Send to unverified users. Should only be used to send invites.
+   * @throws `EmailNotVerifiedError` - If the user's email is not verified or has previously bounced or complained.
+   * @throws `MissingEmailAddressError` - If the user does not have an email address.
+   */
+  async sendEmail({ userId, subject, text }: EmailOptions, force = false) {
```

Comment:
mailer now takes a user id instead of an email. This let's it check whether the user's email is verified before we send the email